Update docs on Module new APIs.

shoumikhin opened the pull request:
Differential Revision: D64005568

dbort commented on docs/source/extension-module.md:
```diff
@@ -94,47 +95,123 @@ if (method_meta.ok()) {
+
+#### Setting Outputs
+
+Currently, only tensor outputs are supported for setting, and they must not be memory-planned at model export time. Memory-planned tensors are preallocated during model export and cannot be replaced.
```
This would be more clear:

"Only outputs of type Tensor can be set at runtime, and ..."

When I first read the current version I interpreted it as "Only outputs of tensors" (which doesn't really mean anything) rather than "only outputs that are tensors".

shoumikhin committed:
Update docs on Module new APIs. (#5952)

Summary: Pull Request resolved: https://github.com/pytorch/executorch/pull/5952

Reviewed By: kirklandsign

Differential Revision: D64005568

shoumikhin commented:
@pytorchbot cherry-pick --onto release/0.4 -c docs

shoumikhin commented:
@pytorchbot cherry-pick --onto release/0.4 -c docs
